### PR TITLE
Hardsuits now block defibulators

### PR DESCRIPTION
**Pull request by beck-thompson**
## About the PR

Title! Defibs now are blocked by hardsuits.

## Why / Balance

I asked in the discord, majority of people seemed to want it. It makes sense and will make medical have to think about defibbing fully armored security.

Also, requires:
https://github.com/space-wizards/space-station-14/pull/31147
To be merged before it works! (I'll also add comments and localization stuff when that happens as I want it fully merged before doing testing and stuff)

## Media

todo

## Requirements

- [ ] I have tested all added content and changes.
- [ ] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl:
- add: Hardsuits now block defibrillators

### CODE REVIEW

**Comment by BlitzTheSquishy** [comment]
Gonna need to make sure that softsuits, (Prisoner, Emergancy, Basic, Clown, Mime, Syndicate) dont, and yes clown and mime are softsuits, not hardsuits (despite their name)

**Comment by Lyndomen** [comment]
💯💯

**Comment by beck-thompson** [comment]
> Gonna need to make sure that softsuits, (Prisoner, Emergancy, Basic, Clown, Mime, Syndicate) dont, and yes clown and mime are softsuits, not hardsuits (despite their name)

Sounds good, when the required PR actually gets merged I'll do more testing to make sure it works on the right things and stuff

**Comment by BlitzTheSquishy** [comment]
> > Gonna need to make sure that softsuits, (Prisoner, Emergancy, Basic, Clown, Mime, Syndicate) dont, and yes clown and mime are softsuits, not hardsuits (despite their name)
> 
> Sounds good, when the required PR actually gets merged I'll do more testing to make sure it works on the right things and stuff

Also currently where that's placed- it looks like EVERY outer slot item will block defib (would include vox tank harnesses i believe)

also you should be testing it BEFORE it gets merged

**Comment by beck-thompson** [comment]
> also you should be testing it BEFORE it gets merged

How? I mean I guess I could like chery pick the required PR ahead of time but I personally don't think that's worth the effort. I mostly made this early so people could give their opinions. If people don't want it, then I wont bother putting the work in finding all the weird hardsuit edge cases

**Comment by BlitzTheSquishy** [comment]
> > also you should be testing it BEFORE it gets merged
> 
> How? I mean I guess I could like chery pick the required PR ahead of time but I personally don't think that's worth the effort. I mostly made this early so people could give their opinions. If people don't want it, then I wont bother putting the work in finding all the weird hardsuit edge cases

pull the merge to your dev or make a dev enviro on wizden (which its already merged) and test there, cause we use similar stuff so

File: Content.Shared/DeltaV/BlockDefibrillator/BlockDefibrillatorComponent.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using Robust.Shared.GameStates;
+
+namespace Content.Shared.DeltaV.BlockDefibrillator;
+
+[RegisterComponent, NetworkedComponent]
```
Comment (MilonPL) [inline Content.Shared/DeltaV/BlockDefibrillator/BlockDefibrillatorComponent.cs:5]: access

File: Content.Shared/DeltaV/BlockDefibrillator/BlockDefibrillatorComponent.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+public sealed partial class BlockDefibrillatorComponent : Component
+{
+
+    [DataField]
+    public string OnBlockPopupMessage = "The defibrillator cannot find a pulse through the armor!";
```
Comment (MilonPL) [inline Content.Shared/DeltaV/BlockDefibrillator/BlockDefibrillatorComponent.cs:10]: locid + localization

File: Content.Shared/DeltaV/BlockDefibrillator/BlockDefibrillatorSystem.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿using Content.Shared.Popups;
+
+namespace Content.Shared.DeltaV.BlockDefibrillator;
+
+public sealed class BlockDefibrillatorSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
```
Comment (MilonPL) [inline Content.Shared/DeltaV/BlockDefibrillator/BlockDefibrillatorSystem.cs:7]: just _popup

File: Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml [hunk 4]
```diff
@@ -146,6 +146,7 @@
       flatReductions:
         Heat: 10 # the average lightbulb only does around four damage!
     slots: OUTERCLOTHING
+  - type: BlockDefibrillator // Delta-V
```
Comment (MilonPL) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml:149]: pro

**Review by MilonPL — CHANGES_REQUESTED** [review]

File: Content.Shared/DeltaV/BlockDefibrillator/BlockDefibrillatorComponent.cs (hunk 1 above)
Comment (beck-thompson) [inline Content.Shared/DeltaV/BlockDefibrillator/BlockDefibrillatorComponent.cs:5]: Add comments

File: Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml (hunk 4 above)
Comment (beck-thompson) [inline Resources/Prototypes/Entities/Clothing/OuterClothing/base_clothingouter.yml:149]: Remember to double check all the weird edge cases for hardsuits. There are a lot :(

**Comment by beck-thompson** [comment]
This isn't too hard to fix but I have more interesting things I'd rather work on! Anyone feel free to copy / re-open this (The code isn't very good so I wouldn't recommend it but whatever 😆)
